{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/sections/templatetags/sections.py

```diff
@@ -12,26 +10,18 @@
 @library.global_function
 @jinja2.contextfunction
 def render_section(context, page_section):
-    try:
-        context = dict(context)
-        context['section'] = page_section
-
-        return render_to_string(f'sections/types/{page_section.template["folder"]}/{page_section.template["file_name"]}', context)
-    except TemplateDoesNotExist as e:
-        import os
-        from django.conf import settings
+    cached_content = cache.get(page_section.cache_key)
+    request = context['request']
 
-        if settings.DEBUG:
-            try:
-                os.makedirs(f'{{ cookiecutter.package_name }}/apps/sections/templates/sections/types/{page_section.template["folder"]}')
-            except OSError as exc:  # Python >2.5
-                if exc.errno == errno.EEXIST and os.path.isdir(page_section.template["folder"]):
-                    pass
-                else:
-                    raise
+    if cached_content and not request.GET.get('cache_bypass') and not request.user.is_staff:
+        return cached_content
 
-            os.system(f'touch {{ cookiecutter.package_name }}/apps/sections/templates/sections/types/{page_section.template["folder"]}/{page_section.template["file_name"]}')
+    context = dict(context)
+    context['section'] = page_section
 
-            return ''
-        else:
-            raise e
+    cached_content = render_to_string('sections/types/{}/{}'.format(
```

Comment:
if the template/folder doesn't exist won't this throw an error? I know this shouldn't happen but it has recently, for example, on GPA with the text-centered section not having a template.

Reply:
Yep, and in my opinion, it _should_ throw an exception.

Reply:
Just making sure we don't mind this throwing an error